skimage/transform/_thin_plate_splines.py

```diff
@@ -54,11 +51,20 @@ def __call__(self, x, y):
                [2., 2., 2., 2., 2.],
                [1., 1., 1., 1., 1.]])
         """
-        if self.parameters is None :
+        if self.parameters is None:
             raise ValueError(f"{self.parameters}. Compute the `estimate`")
-        coeffs =  self.parameters
-        transformed_x = self._transform_points(x, y, coeffs[:, 0])
-        transformed_y = self._transform_points(x, y, coeffs[:, 1])
+        coeffs = self.parameters
+        coords = np.array(coords)
+
+        if coords.ndim == 2:
+            x = coords[:, 0]
+            y = coords[:, 1]
+        else:
+            x = coords[0]
+            y = coords[1]
+
+        transformed_x = self._spline_function(x, y, coeffs[:, 0])
+        transformed_y = self._spline_function(x, y, coeffs[:, 1])
         return [transformed_x, transformed_y]
```

Comment:
The return value should be a NumPy array with the shape `(N, 2)` to be consistent with the other classes. Basically the form and types of the inputs outputs needs to be the same as `_GeometricTransform`. Though, adding more parameters to the input is possible.

Reply:
> The return value should be a NumPy array with the shape (N, 2) to be consistent with the other classes.

@decorouz, in case you missed it, I think this bit isn't addressed yet. :)

Reply:
Thanks for addressing this,